sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/PageBlobAsyncClient.java

```diff
@@ -553,6 +553,20 @@
             .map(rb -> new SimpleResponse<>(rb, rb.getDeserializedHeaders().getCopyStatus()));
     }
 
+    /**
+     * Get the page blob name.
+     *
+     * @return The name of the page blob.
+     */
+    public String getName() {
+        try {
+            return URLParser.parse(new URL(this.azureBlobStorage.getUrl())).getBlobName();
+        } catch (MalformedURLException e) {
```

Comment:
Now that I look at this, it may be smarter to just make an overload `URLParser::parse(String, ClientLogger)` that throws our runtime exception isntead of an extra helper somewhere else.

Reply:
Done.